packages/web-app-files/tests/unit/components/SideBar/Shares/FileShares.spec.ts

```diff
@@ -0,0 +1,224 @@
+import { mount, shallowMount } from '@vue/test-utils'
+import Vuex from 'vuex'
+import FileShares from 'web-app-files/src/components/SideBar/Shares/FileShares.vue'
+import { defaultStubs } from 'web-test-helpers/src/mocks/defaultStubs'
+import { defaultStoreMockOptions } from 'web-test-helpers/src/mocks/store/defaultStoreMockOptions'
+import { defaultComponentMocks } from 'web-test-helpers/src/mocks/defaultComponentMocks'
+import { mockDeep } from 'jest-mock-extended'
+import { Resource } from 'web-client'
+import { SpaceResource } from 'web-client/src/helpers'
+import { defaultLocalVue } from 'web-test-helpers/src/localVue/defaultLocalVue'
+import { v4 as uuidV4 } from 'uuid'
+import { Share, ShareTypes } from 'web-client/src/helpers/share'
+
+const localVue = defaultLocalVue()
+
+const getCollaborator = () => ({
+  shareType: 0,
+  id: uuidV4(),
+  collaborator: {
+    name: 'einstein',
+    displayName: 'Albert Einstein',
+    additionalInfo: 'einstein@example.org'
+  },
+  owner: {
+    name: 'admin',
+    displayName: 'Admin',
+    additionalInfo: 'admin@example.org'
+  },
+  fileOwner: {
+    name: 'admin',
+    displayName: 'Admin',
+    additionalInfo: 'admin@example.org'
+  },
+  file: {},
+  stime: '1639152810',
+  permissions: 15,
+  path: "/Neuer Ordner-'singe'",
+  key: 'collaborator-f5c28709-b921-4ec8-b39a-4c243709b514'
+})
```

Comment:
@pascalwengerter raised a valid concern in https://github.com/owncloud/web/issues/8089

It is nice to have a central place to look at data structures and having to modify them only once should they ever change...

Does it make sense to keep this in a more central place?

Reply:
Yeah, I mean, the place would be our interfaces, no?

The collaborators are a special case right now though. AFAIK we don't have any interface for them. Currently, they look like a hot mess, definitely something to improve in the future.